odoo/http.py

```diff
@@ -528,7 +528,7 @@ def response_wrap(*args, **kw):
                     _logger.info("<function %s.%s> called ignoring args %s" % (f.__module__, f.__name__, ', '.join(ignored)))
 
             response = f(*args, **kw)
-            if isinstance(response, Response) or f.routing_type == 'json':
+            if isinstance(response, Response) or request._request_type == 'json':
```

Comment:
Suggestion, (to validate with @odony or @antonylesuisse)

But if I'm not wrong, the main purpose of this feature type="*" is to support call in http on a json controller.
So basically, be able todo a $.get("/route.json") and receive a str(dict)

We should probably handle the case where your controller return a dict (json case) and that you call the route in http on a route *.

```diff
diff --git odoo/http.py odoo/http.py
index 85388a8589d..edc861580d0 100644
--- odoo/http.py
+++ odoo/http.py
@@ -534,6 +534,9 @@ def route(route=None, **kw):
             if isinstance(response, (bytes, str)):
                 return Response(response)
 
+            if request.endpoint.routing['type'] == "*" and isinstance(response, dict) and request._request_type == 'http':
+                return Response(str(response))
+
             if isinstance(response, werkzeug.exceptions.HTTPException):

```

And from this way, avoid the exception when you call in http a controller '*' that return a dict
`TypeError: 'dict' object is not callable - - -
`



Example:

```python
    @http.route('/controller/star', type='*', auth="public")
    def website_star(self, **kwargs):
        return {
            'feature1': kwargs.get('value1'),
            'feature2': kwargs.get('value2'),
            'version': odoo.service.common.exp_version()
        }
```

```js
>> odoo.__DEBUG__.services['web.ajax'].jsonRpc('/controller/star', 'call', {value1:'val1', value2:'val2'}).then(o=>console.log(o))
```
> {"jsonrpc": "2.0", "id": 918268530, "result": {"feature1": "val1", "feature2": "val2", "version": {"server_version": "13.4alpha1", "server_version_info": [13, 4, 0, "alpha", 1, ""], "server_serie": "13.4", "protocol_version": 1}}}

```js
>> $.get('/controller/star?value1=val1&value2=val2').then(o=>console.log(o));
```

> {'feature1': 'val1', 'feature2': 'val2', 'version': {'server_version': '13.4alpha1', 'server_version_info': (13, 4, 0, 'alpha', 1, ''), 'server_serie': '13.4', 'protocol_version': 1}}

Reply:
@JKE-be Indeed, good idea :)

I just call `json.dumps` instead of `str` because single quotes are not valid in a JSON :)